src/InteractsWithMessenger.php

```diff
@@ -33,15 +36,8 @@ final protected static function _resetMessengerTransports(): void
 
     final protected function messenger(?string $transport = null): TestTransport
     {
-        if (!$this instanceof KernelTestCase) {
-            throw new \LogicException(\sprintf('The %s trait can only be used with %s.', __TRAIT__, KernelTestCase::class));
-        }
-
-        if (!self::$booted) {
-            self::bootKernel();
-        }
-
-        $container = \method_exists($this, 'getContainer') ? self::getContainer() : self::$container;
+        $this->init();
+        $container = self::getContainer();
```

Comment:
about deprecation, I guess you'll have to rename this method `transport()`

then you could do something like this:
```php
/**
 * @deprecated use transport() instead
 */ 
final protected function messenger(?string $transport = null): TestTransport
{
    trigger_deprecation('zenstruck\messenger-test', '1.7.0', '"messenger()" method is deprecated and will be removed in 2.0. Please use "transport()" instead.');

    return $this->transport($transport);
}
```

Reply:
Thanks for this advice